### PR TITLE
docs: add '/v2/farcaster/user/memberships/list' endpoint

### PR DESCRIPTION
**Pull request by eurbs**
Adds documentation for the new `/v2/farcaster/user/memberships/list` endpoint

### CODE REVIEW

**Review by rishavmukherji — APPROVED** [review]

File: src/v2/spec.yaml [hunk 1]
```diff
@@ -4045,6 +4045,52 @@ paths:
...
+            minimum: 1
+            maximum: 100
+        - name: cursor
+          in: query
+          description: Pagination cursor.
+          required: false
+          schema:
+            type: string
+      responses:
+        "200":
```
Comment (Shreyaschorge) [inline src/v2/spec.yaml:4082]: Need API change
Cursor should be `response.next.cursor`. 
image

File: src/v2/spec.yaml [hunk 2]
```diff
@@ -4045,6 +4045,52 @@ paths:
...
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ChannelMemberListResponse"
```
Comment (Shreyaschorge) [inline src/v2/spec.yaml:4087]: In `ChannelMemberListResponse` Schema `next` should be in the required array.

Comment (Shreyaschorge) [inline src/v2/spec.yaml:4087]: `ChannelMember` Schema should have a required property called object with `member` a enum

File: src/v2/spec.yaml [hunk 3]
```diff
@@ -4045,6 +4045,52 @@ paths:
...
+        "400":
+          $ref: "#/components/responses/400Response"
+        "404":
```
Comment (Shreyaschorge) [inline src/v2/spec.yaml:4090]: API is not sending a 404 response

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

File: src/v2/spec.yaml (hunk 1 above)
Comment (eurbs) [reply src/v2/spec.yaml:4082]: thanks good catch! https://github.com/neynarxyz/api/pull/881

File: src/v2/spec.yaml (hunk 2 above)
Comment (eurbs) [reply src/v2/spec.yaml:4087]: good call

File: src/v2/spec.yaml (hunk 3 above)
Comment (eurbs) [reply src/v2/spec.yaml:4090]: oh interesting, i would have thought `validate_fid()` middleware throws a 404 if not found, but I'm wrong about that

File: src/v2/spec.yaml (hunk 1 above)
Comment (manan19) [reply src/v2/spec.yaml:4089]: nice catch @Shreyaschorge

**Pull request merged**